curricula/views.py

```diff
@@ -388,43 +389,39 @@ def unit_compiled(request, slug, unit_slug):
 
 # @login_required
 def unit_pdf(request, slug, unit_slug):
+    try:
+        buffer = StringIO()
+        c = pycurl.Curl()
+        c.setopt(c.WRITEDATA, buffer)
 
-    buffer = StringIO()
-    c = pycurl.Curl()
-    c.setopt(c.WRITEDATA, buffer)
-
-    unit = get_object_or_404(Unit, curriculum__slug=slug, slug=unit_slug)
+        unit = get_object_or_404(Unit, curriculum__slug=slug, slug=unit_slug)
 
-    try:
         c.setopt(c.URL, get_url_for_pdf(request, unit.get_compiled_url(), True))
         print unit.get_compiled_url()
         c.perform()
 
         c.close()
         compiled = buffer.getvalue()
-    except Exception:
-        logger.exception('PDF Curling Failed')
-        return HttpResponse('PDF Curling Failed', status=500)
 
-    if request.GET.get('html'):  # Allows testing the html output
-        response = HttpResponse(compiled)
-    else:
-        try:
+        if request.GET.get('html'):  # Allows testing the html output
+            response = HttpResponse(compiled)
+        else:
             pdf = pdfkit.from_string(compiled.decode('utf8'), False, options=settings.WKHTMLTOPDF_CMD_OPTIONS, configuration=pdfkit_config)
-        except Exception:
-            logger.exception('PDF Generation Failed')
-            return HttpResponse('PDF Generation Failed', status=500)
 
-        response = HttpResponse(pdf, content_type='application/pdf')
-        response['Content-Disposition'] = 'inline;filename=unit%s.pdf' % unit.number
+            response = HttpResponse(pdf, content_type='application/pdf')
+            response['Content-Disposition'] = 'inline;filename=unit%s.pdf' % unit.number
 
-        ip = get_real_ip(request)
-        slack_message('slack/message.slack', {
-            'message': 'created a PDF from %s %s' % (slug, unit_slug),
-            'user': request.user.get_username() or ip,
-        })
+            ip = get_real_ip(request)
+            slack_message('slack/message.slack', {
+                'message': 'created a PDF from %s %s' % (slug, unit_slug),
+                'user': request.user.get_username() or ip,
+            })
+            return response
+    except Exception, e:
```

Comment:
Nitpick, but let's change this to `RuntimeError` - usually that's as generic as you should go with catching exceptions unless you're doing something weird. See all the stuff we don't want to catch: https://docs.python.org/2/library/exceptions.html#exception-hierarchy

Reply:
Maybe I remembered wrong on this - I guess `Exception` is fine as long as it's not just `except:` by itself, which does catch things like `KeyboardInterrupt`. I think this is fine.

Reply:
Nonetheless, thanks for pointing this out - I was not consciously aware of how the python exception hierarchy looks like and was going along with the precedent set by the above code. This will be useful information to be aware of for the future.